### ISSUE TITLE
i18n

### ISSUE THREAD

**Issue by mist** [michaelpelletier/ReadyUpChummer#1]
Ahoi,
I just found this generator and I like it a lot. I just have the problem that is in English and I would like to translate it to German. Is there any nice way to do this? Like a language file with all the strings in it.

Also do you plan on working on it? And is help from outside welcome?

Cheers,
mist

**Comment by michaelpelletier** [comment]
Oh man, I haven't touched this in ages, and never actually finished it. (Man, some of that code looks bad). 

Given how there are different ways than the priority system now with Run Faster, I'm not even sure that this is even still relevant. You're definitely welcome to play around with it, but I'm not sure if I'll end up getting back into it. The amount of text that I've put in here (probably) violates copyright, too - I was only really planning to use this for my own players.

A language file could definitely work for things that are in the javascript, and you could import the language as global variables or something that way. Maybe an object literal, so that you could replace each instance of something like label: "Bod" as being something more like - label: language.bodyShortLabel.

For things that are included in the page, like table headers, I'd probably put both in and have JS toggle them. Eg: <div ng-lang="en">Column 1</div><div ng-lang="de">Column 1</div>, and then have the Javascript hide/show those based on what language is selected. (Like http://stackoverflow.com/questions/13606290/html-change-language-using-element-lang)

**Comment by mist** [comment]
Yeah, I realized that, I thought maybe I could work on the logic as well.

I still don't know anything else than the priority system, `Run Faster` is afaik not translated yet. But because I guess it will be translated and you say the priority system is not needed anymore it would be non-needed work for me to work on this. So I guess I won't do it.

I guess the best way would be to switch all strings to javascript to be able to use a language file, the version with all versions inside the HTML will be very bloaty when adding more than one language.

I think I will wait until run faster is out to decide if I want to do anything here.
If you don't mind though I will probably take your HTML/CSS-Design to build a nice character manager without any smart logic - similar to my [sr5-charsheetgen repo](https://github.com/mist/sr5-charsheetgen) with an [PDF output](https://github.com/mist/sr5-charsheetgen/blob/mist/Timur_Kowaljow/chargen.pdf?raw=true)
